### PR TITLE
Update for MediaPlaceholder component: don't stack error messages when re-uploading

### LINKED ISSUE #6106: Uploading messages for media shouldn't stack
The following is what happens after a series of upload failures:

audioerrormessagesstack-bug

Ideally the messages shouldn't stack like this, the last one should stay and be replaced as you get another.

### PR DESCRIPTION
**Pull request by v9ai**
## Description
Fixes: https://github.com/WordPress/gutenberg/issues/6106
The implementation is based on this stale PR's feedback: https://github.com/WordPress/gutenberg/pull/10250

## How has this been tested?
It was tested locally.

## Screenshots 
Peek 2019-03-30 17-28

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by nicolad** [comment]
Hello @noisysocks, taken into account your feedback from here: https://github.com/WordPress/gutenberg/pull/10250
I've tried to fix the issue in 2 blocks.
Please confirm if that's ok and then I will add that fix in other blocks which use `MediaPlaceholder`.

File: packages/block-library/src/audio/edit.js [hunk 1]
```diff
@@ -98,9 +99,15 @@ class AudioEdit extends Component {
 		this.setState( { editing: false } );
 	}
 
+	onUploadError( message ) {
```
Comment (nicolad) [inline packages/block-library/src/audio/edit.js]: I've placed this method in alphabetic order, please confirm if that's good.

**Review by Soean — DISMISSED** [review]
Thanks for the PR, it works great in the audio block. We should do the same in the other blocks with upload option: Gallery, Media & Text, Video, Cover, File.

File: packages/block-library/src/media-text/media-container.js [hunk 2]
```diff
@@ -81,6 +92,8 @@ class MediaContainer extends Component {
 				onSelect={ onSelectMedia }
 				accept="image/*,video/*"
 				allowedTypes={ ALLOWED_MEDIA_TYPES }
+				notices={ noticeUI }
```
Comment (nicolad) [inline packages/block-library/src/media-text/media-container.js]: `noticeUI` is not fitting properly here, should be addressed in a separate issue:
image

**Comment by nicolad** [comment]
Thank you @Soean for taking time and reviewing this.
I have adjusted and tested locally all components you've mentioned.

**Comment by nicolad** [comment]
@gziolo, may I also have your review on this?

File: packages/block-library/src/audio/edit.js [hunk 3]
```diff
@@ -98,13 +99,20 @@ class AudioEdit extends Component {
 		this.setState( { editing: false } );
 	}
 
+
+	onUploadError( message ) {
+		const { noticeOperations } = this.props;
+		noticeOperations.removeAllNotices();
+		noticeOperations.createErrorNotice( message );
+  }
```
Comment (aduth) [inline packages/block-library/src/audio/edit.js]: The spaces here ought to be a tab.

**Review by aduth — COMMENTED** [review]
Thanks for your patience in the review here. While there's separate ongoing discussion about how `withNotices` should be implemented or if it should be used (e.g. #9442), this seems like a sensible improvement on the current iteration of notices use in these components.

File: packages/block-library/src/audio/edit.js (hunk 3 above)
Comment (nicolad) [reply packages/block-library/src/audio/edit.js]: Fixed. thx.

**Comment by nicolad** [comment]
> Thanks for your patience in the review here. While there's separate ongoing discussion about how `withNotices` should be implemented or if it should be used (e.g. #9442), this seems like a sensible improvement on the current iteration of notices use in these components.

Thanks for letting me know this, I've subscribed to that discussion and related PRs in order to be on the same page and to apply the final solution when it will be available.

**Review by aduth — APPROVED** [review]
Thanks for the work here. Tested across all of the impacted blocks and appears to work well 👍

**Pull request merged**